xonsh/tools.py

```diff
@@ -909,3 +909,50 @@ def check_for_partial_string(x):
         return (string_indices[-1], None, starting_quote[-1])
     else:
         return (string_indices[-2], string_indices[-1], starting_quote[-1])
+
+
+# expandvars is a modified version of os.path.expandvars from the Python 3.5.1
+# source code (root/Lib/posixpath.py, line 266)
+
+_varprog = None
+
+
+def expandvars(path):
+    """Expand shell variables of form $var and ${var}.  Unknown variables
+    are left unchanged."""
+    global _varprog
+    ENV = builtins.__xonsh_env__
+    if isinstance(path, bytes):
+        path = path.decode(encoding=env.get('XONSH_ENCODING'),
+                           errors=env.get('XONSH_ENCODING_ERRORS'))
+    if '$' not in path:
+        return path
+    if not _varprog:
+        import re
+        _varprog = re.compile(r'\$(\w+|\{[^}]*\})', re.ASCII)
```

Comment:
This doesn't handle `%name%` completion on windows, which someone will probably complain about :) https://docs.python.org/3.5/library/os.path.html#os.path.expandvars  Shouldn't be too hard to modify.